doc/rst/language/spec/classes.rst

```diff
@@ -622,10 +622,19 @@ Overriding Base Class Methods
 
 If a method in a derived class is declared with a signature identical to
 that of a method in a base class, then it is said to override the base
-class’s method. Such methods are considered for dynamic dispatch. In
-particular, dynamic dispatch will be used when the method receiver has a
-static type of the base class but refers to an instance of a derived
-class type.
+class’s method. Such methods may be considered for dynamic dispatch if
+certain criteria are met. In particular, dynanmic dispatch will be used
+when the method receiver has a static type of the base class but refers
+to an instance of a derived class type. Additionally, a method eligible
+for dynamic dispatch must not be a class method (see :ref:`Class_Methods`),
+must not return ``type``, and must not return ``param``.
+
+   *Rationale*.
+
+   Class methods, methods that return ``type``, and methods that return
+   ``param`` are not considered as candidates for dynamic dispatch because
+   they are resolved at compile-time based on the static type of the
+   method receiver.
 
 In order to have identical signatures, two methods must have the same
 the names, intents, types, and order of formal arguments. The return
```

Comment:
While you're here, there's an extra "the" here.

Assuming "names" here refers to the arguments, then I don't think anything in the definition of overriding restricts it to methods that have the same name.  The paragraph above says it applies to two methods that have identical signatures, and this definition of identical signatures doesn't mention the names of the methods themselves.

Reply:
That's a really good point. 

Assuming we're taking signature to mean the _type_ of the function and not the name, I'll have to make some adjustments.

Better double check the spec first just to make sure, though.